gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/messages/GfxdSystemProcedureMessage.java

```diff
@@ -1062,7 +1066,65 @@ String getSQLStatement(Object[] params) throws StandardException {
             .append(params[2]).append(')').toString();
       }
     },
-    
+
+    setLogLevel {
+
+      @Override
+      boolean allowExecution(Object[] params) {
+        // so allowing log level to be set on all nodes including locator
+        return true;
+      }
+
+      @Override
+      public void processMessage(Object[] params, DistributedMember sender)
+          throws StandardException {
+        try {
+          String logClass = (String) params[0];
+          Level level = org.apache.log4j.Level.toLevel((String) params[1]);
+          SanityManager.DEBUG_PRINT(GfxdConstants.TRACE_SYS_PROCEDURES,
+                  "GfxdSystemProcedureMessage: setting log level for class '" + logClass
+                          + "' to " + level);
+          if (logClass.equals("")) {
+            // sets the log level for the root logger and the GFXD bridge
+            LogManager.getRootLogger().setLevel(level);
+            if (InternalDistributedSystem.getLoggerI18n() instanceof GFToSlf4jBridge) {
+              ((GFToSlf4jBridge) InternalDistributedSystem.getLoggerI18n()).setLevelForLog4jLevel(level);
```

Comment:
Call getLoggerI18n only once. The more normal way in GFXD layer is to use "Misc.getCacheLogWriterNoThrow()".